lib/http.js

```diff
@@ -24,6 +24,7 @@ function httpRequest(sdk, options) {
       args = options.args,
       saveAuthnState = options.saveAuthnState,
       accessToken = options.accessToken,
+      withCredentials = options.withCredentials === false ? false : true,
```

Comment:
You have this pattern a few places - is there a specific reason you aren't using `!!options.withCredentials;`?

Reply:
The code is not allowing `withCrendentials = undefined` to actually disable auth headers, it is forcing it to be explicitly set as false.

I don't remember, but my guess is this is working around the behavior of `util.extend` and values of 'undefined'. I will investigate further

Reply:
But `!!options.withCredentials` will become `false`, even if it was `undefined`.

Reply:
This is because 'withCredentials' should be `true` by default. So if it is 'undefined', it will still be true.